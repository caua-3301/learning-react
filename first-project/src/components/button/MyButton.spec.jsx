import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { MyButton } from '.';

describe('<MyButton />', () => {
  it('should content and click', () => {
    const fn = jest.fn();

    //renderiza um componente no ambiente de testes (simula como a exbição na web)
    render(<MyButton disabled={false} onClick={fn} />);

    //debug();

    const botao = screen.getByRole('button', { name: 'Load more' });

    userEvent.click(botao);

    //expect(fn).toHaveBeenCalledTimes(1);

    expect(botao).toBeInTheDocument();
  });

  it('should make the snapshot', () => {
    const fn = jest.fn();

    //renderiza um componente no ambiente de testes (simula como a exbição na web)
    render(<MyButton text="Load more and more" disabled={true} onClick={fn} />);
    const buttonElement = screen.getByRole('button', { name: 'Load more' });

    expect(buttonElement).toMatchSnapshot();
  });
});
