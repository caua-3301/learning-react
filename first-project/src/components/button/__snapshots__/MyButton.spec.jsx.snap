// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyButton /> should make the snapshot 1`] = `
<button
  disabled=""
>
  Load more
</button>
`;
